src/runtime-rs/crates/resource/src/rootfs/nydus_rootfs.rs

```diff
@@ -0,0 +1,152 @@
+            NYDUS_ROOTFS_V5 => {
+                let rafs_mnt =
+                    do_get_guest_share_path(HYBRID_ROOTFS_WRITABLE_LAYER_LOWER_DIR, cid, true);
+                rafs_mount(
+                    h,
+                    rafs_meta.to_string(),
+                    rafs_mnt,
+                    extra_options.config.clone(),
+                    None,
+                )
+                .await
+                .context("failed to do rafs mount")?;
+                let container_share_dir = get_host_rw_shared_path(sid)
+                    .join(PASSTHROUGH_FS_DIR)
+                    .join(cid);
+                let rootfs_dir = container_share_dir.join(ROOTFS);
+                fs::create_dir_all(rootfs_dir).context("failed to create directory")?;
+                let mount_result = share_fs_mount
+                    .share_rootfs(ShareFsRootfsConfig {
+                        cid: cid.to_string(),
+                        source: extra_options.snapshot_dir.clone(),
+                        target: SNAPSHOT_DIR.to_string(),
+                        readonly: true,
+                        is_rafs: false,
+                    })
+                    .await
+                    .context("share nydus rootfs")?;
+                let mut options: Vec<String> = Vec::new();
+                options.push(
+                    "lowerdir=".to_string()
+                        + &do_get_guest_path(
+                            HYBRID_ROOTFS_WRITABLE_LAYER_LOWER_DIR,
+                            cid,
+                            false,
+                            true,
+                        ),
+                );
+                options.push(
+                    "workdir=".to_string()
+                        + &do_get_guest_path(
+                            format!("{}/{}", SNAPSHOT_DIR, "work").as_str(),
+                            cid,
+                            false,
+                            false,
+                        ),
+                );
+                options.push(
+                    "upperdir=".to_string()
+                        + &do_get_guest_path(
+                            format!("{}/{}", SNAPSHOT_DIR, "fs").as_str(),
+                            cid,
+                            false,
+                            false,
+                        ),
+                );
+                options.push("index=off".to_string());
+                Ok((
+                    Storage {
+                        driver: KATA_OVERLAY_DEV_TYPE.to_string(),
+                        source: TYPE_OVERLAY_FS.to_string(),
+                        fs_type: TYPE_OVERLAY_FS.to_string(),
+                        options,
+                        mount_point: mount_result.guest_path.clone(),
+                        ..Default::default()
+                    },
+                    mount_result.guest_path,
+                ))
+            }
+            NYDUS_ROOTFS_V6 => {
```

Comment:
IIUC both v5 and v6 can be handled by the builtin nydus support in dragonball. The only difference is that v6 can be handled by the guest kernel as well. So let's just handle both v5/v6 in dragonball first. And once the kernel part is ready, we can add an option to let runtime-rs handle nydus v6 in the guest kernel optionally.

Reply:
Sure, I will put the nydus v6 rootfs support for dragonball in the next pr